pkg/db/db_common/server_settings.go

```diff
@@ -0,0 +1,190 @@
+package db_common
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"reflect"
+	"time"
+
+	"github.com/jackc/pgx/v5"
+	"github.com/spf13/viper"
+	"github.com/turbot/steampipe/pkg/constants"
+	"github.com/turbot/steampipe/pkg/utils"
+	"github.com/turbot/steampipe/sperr"
+)
+
+type ServerSettingKey string
+type ServerSettings struct {
+	StartTime        time.Time
+	SteampipeVersion string
+	FdwVersion       string
+	CacheMaxTtl      int
+	CacheMaxSizeMb   int
+	CacheEnabled     bool
+
+	// a private property
+	// defaults to false - set to true after loading completes
+	loaded bool
+}
+
+// StubServerSettings returns a server settings struct which is maked as unloaded
+func StubServerSettings() *ServerSettings {
+	return new(ServerSettings)
+}
+
+func LoadServerSettings(ctx context.Context, conn *pgx.Conn) (*ServerSettings, error) {
+	rows, err := conn.Query(ctx, fmt.Sprintf("SELECT name,value FROM %s.%s", constants.InternalSchema, constants.ServerSettingsTable))
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+	settings := new(ServerSettings)
+	for rows.Next() {
+		var name ServerSettingKey
+		var value any
+		if err := rows.Scan(&name, &value); err != nil {
+			return nil, sperr.WrapWithMessage(err, "error reading row from %s.%s", constants.InternalSchema, constants.ServerSettingsTable)
+		}
+
+		switch name {
+		case ServerSettingSteampipeVersion:
```

Comment:
why not use custom tags on the server setting struct

```
type ServerSettings struct {
	StartTime        time.Time `column:"fdw_version,time"`
	SteampipeVersion string `column:"steampipe_version,text"`
	FdwVersion       string      `column:"fdw_version,text"`
	CacheMaxTtl      int          `column:"cache_max_ttl,int"`
	CacheMaxSizeMb   int.     etc
	CacheEnabled     bool

}

```
Then when reading rows, you can iterate through the properties of ServerSettings to find a matching tag, the switch on the tag type to set the property value:

```
switch tagType {
  case "text":
       setProperty(p, value.(string))
 case "int":
       setProperty(p, value.(int))
}
etc
```

}

Reply:
we do similar stuff for introspection columns